### PR TITLE
refactor: Mmr tests no longer depend on storage layer

### PR DESCRIPTION
**Pull request by dan-da**
Breaking Change

* adds src/mock that is built if `mock` feature-flag enabled, and for unit tests
* removes `ArchivalMmr` and replaces with `MockMmr`, in memory
* removes `RustyLevelDbVec`, `DatabaseArray`, `DatabaseVec`
* use MockMmr instead of ArchivalMmr in Mmr tests

### CODE REVIEW

**Review by aszepieniec — APPROVED** [review]

File: twenty-first/src/mock/mmr/mock_mmr.rs [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
-use super::shared_advanced;
-use super::shared_basic;
+use crate::util_types::mmr::{
+    mmr_accumulator::MmrAccumulator, mmr_membership_proof::MmrMembershipProof, mmr_trait::Mmr,
+    shared_advanced, shared_basic,
+};
 
+/// MockMmr is available for feature `mock` and for unit tests.
+///
+/// It implements an in-memory Mmr using a Vec<Digest> as the data store.
```
Comment (Sword-Smith) [inline twenty-first/src/mock/mmr/mock_mmr.rs:15]: You should clarify that this is an archival-MMR, not an MMR accumulator. An archival MMR stores *all* nodes in multiple trees whereas an MMR accumulator only stores the roots of the binary trees.

This clarification could take place either in the documentation, or through the name of the mock struct. Up to you.

File: twenty-first/src/util_types/mmr/mmr_membership_proof.rs [hunk 2]
```diff
@@ -1264,21 +1250,20 @@ mod mmr_membership_proof_test {
         type H = blake3::Hasher;
 
         for leaf_count in 0..68u64 {
-            // 1. Build an ArchivalMmr with a variable amount of leaves
+            // 1. Build an MockMmr with a variable amount of leaves
```
Comment (Sword-Smith) [inline twenty-first/src/util_types/mmr/mmr_membership_proof.rs:1253]: 'a' :)

File: twenty-first/src/mock/mmr/mock_mmr.rs [hunk 3]
```diff
@@ -700,7 +717,7 @@ mod mmr_test {
                         .0
                 );
 
-                // Verify that membership proofs are the same as generating them from an archival MMR
+                // Verify that membership proofs are the same as generating them from an MockMmr
```
Comment (Sword-Smith) [inline twenty-first/src/mock/mmr/mock_mmr.rs:720]: 'a' :)

**Review by Sword-Smith — CHANGES_REQUESTED** [review]
Looks good. I only request a few updates to documentation.

File: twenty-first/src/util_types/mmr/mmr_membership_proof.rs (hunk 2 above)
Comment (dan-da) [reply twenty-first/src/util_types/mmr/mmr_membership_proof.rs]: done.

File: twenty-first/src/mock/mmr/mock_mmr.rs (hunk 3 above)
Comment (dan-da) [reply twenty-first/src/mock/mmr/mock_mmr.rs]: done.

**Pull request merged**